executor/executor_runner.h

```diff
@@ -215,7 +265,13 @@ class Proc
 				while (ReadOutput())
 					;
 			}
-			HandleCompletion(status);
+			bool hanged = SYZ_EXECUTOR_USES_FORK_SERVER && state_ == State::Executing;
+			HandleCompletion(status, hanged);
+			if (hanged) {
+				// If the process has hanged, it may still be using per-proc resources,
+				// so allocate a fresh proc id.
+				effective_id_ = proc_id_pool_.Alloc(effective_id_);
```

Comment:
I wonder what may break if we just overwrite `id_` here and don't split it in two?

Reply:
At least LastExecuting type is not prepared to handle IDs larger than max (will panic). Not sure what else may break if we report IDs larger than max.
But even if we try to fix LastExecuting to handle larger IDs, what should it do? Should it keep last N programs for each unique proc ID? This would result in much larger log with more (unuseful) programs. I think the only way to do it is to still split id into 2 and send both to the host, but that's even more complicated.

Reply:
If it's only about `LastExecuting`, that's IMO not critical -- we can change these rules anytime however we want. E.g. if a proc has hanged, in `LastExecuting` we could discard all of its progs before the hanged one. These hanged progs are already prepended to the crash log in this PR, so the difference in the log size may be insignificant. But it will make our C++ code easier to understand.

Reply:
Done. PTAL